client/player/pulse_player.cpp

```diff
@@ -177,15 +177,32 @@ bool PulsePlayer::needsThread() const
     return true;
 }
 
-
 void PulsePlayer::worker()
 {
-    pa_mainloop_run(pa_ml_, nullptr);
-}
+    while (active_)
+    {
+        pa_mainloop_run(pa_ml_, nullptr);
+
+        // if we are still active, wait for a chunk and attempt to reconnect
+        while (active_ && !stream_->waitForChunk(100ms))
+        {
+            LOG(DEBUG, LOG_TAG) << "Waiting for a chunk to become available before reconnecting\n";
+        }
 
+        if (active_)
+        {
+            LOG(INFO, LOG_TAG) << "Chunk available, reconnecting to pulse\n";
+            this->connect();
```

Comment:
`connect()` can throw an exception, that must be caught here, similar to alsa:

```
try
{
   LOG(INFO, LOG_TAG) << "Chunk available, reconnecting to pulse\n"
   connect();
}
catch (const std::exception& e)
{
   LOG(ERROR, LOG_TAG) << "Exception while connecting to pulse: " << e.what() << endl;
   disconnect();
   chronos::sleep(100);
}
```
Also some members might be initialized in `connect()`, I think a call to `disconnect()` should do the necessary cleanup.

Reply:
I will add it